### ISSUE TITLE
[feature request] SidecarSet add upgrade state in pod annotation

### ISSUE THREAD

**Issue by zmberg** [openkruise/kruise#1312]
**What would you like to be added**:
SidecarSet can publish Sidecar containers in batches, and currently needs to add a representation to make it easier for upper-level paas to support more clearly which Pods are in the process of being published and which are finished. 

Currently Pod have sidecarset annotations[kruise.io/sidecarset-inplace-update-state] and value is struct:
```
type SidecarSetUpgradeSpec struct {
	UpdateTimestamp              metav1.Time `json:"updateTimestamp"`
	SidecarSetHash               string      `json:"hash"`
	SidecarSetName               string      `json:"sidecarSetName"`
	SidecarList                  []string    `json:"sidecarList"`                  // sidecarSet container list
	SidecarSetControllerRevision string      `json:"controllerRevision,omitempty"` // sidecarSet controllerRevision name
}
```

So we can add field in the above struct:
```
type SidecarSetUpgradeSpec struct {
	...
         // enum: Normal, Updating
        State string `json:"state"`
}
```
```

**Comment by xiangpingjiang** [comment]
The value of sidecarset annotations[kruise.io/sidecarset-inplace-update-state] is [here](https://github.com/openkruise/kruise/blob/ffcf1898fd124c0b4adde9cccc11d043ff0d40ef/apis/apps/pub/inplace_update.go#L51).  


The value of sidecarset annotations[kruise.io/sidecarset-hash] is [here](https://github.com/openkruise/kruise/blob/ffcf1898fd124c0b4adde9cccc11d043ff0d40ef/pkg/control/sidecarcontrol/util.go#L75).   

Maybe you have confused these two annotations @zmberg

**Comment by zmberg** [comment]
> 

Yes, you are right. You can update the sidecarset annotations[kruise.io/sidecarset-hash]

**Comment by xiangpingjiang** [comment]
> > 
> 
> Yes, you are right. You can update the sidecarset annotations[kruise.io/sidecarset-hash]

Ok, I will do that recentlly

**Comment by LvWuqian** [comment]
@zmberg Hello, I want to work on this issue, can you assign it to me?
